crates/storage/src/collections/binary_heap/mod.rs

```diff
@@ -165,7 +169,7 @@ where
     /// Returns `None` if the heap is empty
     pub fn pop(&mut self) -> Option<T> {
         // replace the root of the heap with the last element
-        let elem = self.elems.swap_remove(0);
+        let elem = self.elements.swap_remove(0);
```

Comment:
Btw.: Since `self.elements` after this optimization now always stores both children instead of just one: Is this operation still valid in this way? Does it always yield a properly sorted binary heap?

Reply:
Yes, the code doesn't change the way indices work on this `mod.rs` level, only on the levels below something changed. But the abstraction works in a way that there is no difference up here. And `0` is always the root.

But yeah I would also feel better about having more fuzz tests. We already have some for `binary_heap` and I aim to add more in the upcoming PR for more storage primitive testing.